### PR TITLE
Change PartitionedFileSetDataset#dropPartition to recursively delete …

### PR DESCRIPTION
**Pull request by anwar6953**
…the partition files. Also throw dataset exception if the call to delete fails without throwing an exception.

The necessary change is passing `true` to the call to `delete`. The remainder is additional error-checking.

http://builds.cask.co/browse/CDAP-RBT309-4
https://issues.cask.co/browse/CDAP-3019

### CODE REVIEW

**Comment by anwar6953** [comment]
@chtyim The behavior for HDFSLocation is different than LocalLocation when delete() is called without recursive=true and there existing files under that Location.
LocalLocation simply returns `false`, but `HDFSLocation` threw an exception.
Shouldn't this inconsistency not exist? (Would it be a good idea to make LocalLocation also throw an exception in that case)?

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java [hunk 1]
```diff
@@ -330,11 +330,15 @@ public void dropPartition(PartitionKey key) {
       return;
     }
     if (!isExternal) {
+      String deleteFailureMessage = String.format("Error deleting file(s) for partition %s at path %s.",
```
Comment (anew) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: it is better to create the message only if there is an error.

**Comment by chtyim** [comment]
Can you open a Twill issue with the stacktrace attached?

**Comment by anwar6953** [comment]
Created: https://issues.apache.org/jira/browse/TWILL-140

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java (hunk 1 above)
Comment (anwar6953) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java]: ok; done.

**Comment by anwar6953** [comment]
Added an additional check to check for file existence before delete (it's possible the user didn't create a file at that partition's path when creating a partition.

**Comment by anew** [comment]
LGTM

**Pull request merged**